test/module/irohad/multi_sig_transactions/mst_processor_test.cpp

```diff
@@ -294,6 +295,60 @@ TEST_F(MstProcessorTest, onNewPropagationUsecase) {
   propagation_subject.get_subscriber().on_next(peers);
 }
 
+/**
+ * @given initialised mst processor
+ * AND our state contains one transaction
+ *
+ * @when received notification about new propagation
+ * AND transport successfully sent the state
+ *
+ * @then same diff is applied to storage
+ */
+TEST_F(MstProcessorTest, SendStateSuccess) {
+  // ---------------------------------| given |---------------------------------
+  auto quorum = 2u;
+  mst_processor->propagateBatch(addSignaturesFromKeyPairs(
+      makeTestBatch(txBuilder(1, time_after, quorum)), 0, makeKey()));
+  EXPECT_CALL(*transport, sendState(_, _))
+      .WillOnce(Return(rxcpp::observable<>::just(true)));
+
+  // ---------------------------------| when |----------------------------------
+  std::vector<std::shared_ptr<shared_model::interface::Peer>> peers{
+      makePeer("one", another_peer_key_hex)};
+  propagation_subject.get_subscriber().on_next(peers);
+
+  // ---------------------------------| then |----------------------------------
+  ASSERT_TRUE(
+      storage->getDiffState(another_peer_key_hex, time_after).isEmpty());
+}
+
+/**
+ * @given initialised mst processor
+ * AND our state contains one transaction
+ *
+ * @when received notification about new propagation
+ * AND transport failed to send the state
+ *
+ * @then diff is not applied to storage
+ */
+TEST_F(MstProcessorTest, SendStateFailure) {
+  // ---------------------------------| given |---------------------------------
+  auto quorum = 2u;
+  mst_processor->propagateBatch(addSignaturesFromKeyPairs(
+      makeTestBatch(txBuilder(1, time_after, quorum)), 0, makeKey()));
+  EXPECT_CALL(*transport, sendState(_, _))
+      .WillOnce(Return(rxcpp::observable<>::just(false)));
```

Comment:
should there be also be a test with empty observable? in my understanding of the modular test approach it should. I'd rewrite these tests with a fixture then, to have only the `sendState` expectation and state diff check in each `TEST_F`

Reply:
Discussed offline, transport interface is restricted to return exactly one value at the moment, and if the interface changes, then the test will be refactored.